tasks/Functions-admin/manageAllowlist.js

```diff
@@ -43,14 +56,188 @@ async function addOrRemove(action, taskArgs) {
   console.log(`Allowlist updated for oracle ${oracle.address} on ${network.name}`)
 }
 
-task("functions-add-senders", "Add wallets to allowlist in the Oracle contract")
-  .addParam("addresses", "Comma-separated list of addresses")
+task("functions-add-senders", "Add wallets to allowlist in the Oracle contract.  In order to add users from allowlist.csv, copy the CSV file into the root directory and do not set the addresses parameter.")
+  .addOptionalParam("addresses", "Comma-separated list of addresses.  If this is not provided, addresses will be pulled from allowlist.csv")
```

Comment:
Is it possible to `--addresses $(cat allowlist.csv)` and to not deal with allowlist.csv?

Reply:
No.  There is additional functionality that we get by using the whole `allowlist.csv` file instead of a plain list of addresses.  Namely:
1. Validation of event code
2. Validation the user agreed to TOS
3. The command produces an `updatedAllowlist.csv` file with notes indicating the reason a user wasn't added if validation of the address, event code or TOS fails.  It also produces a `invalidUsers.csv` that just shows invalid users.